### PR TITLE
Add max_token_length setting to the CharGroupTokenizer

### PR DESCRIPTION
**Pull request by ADBalici**
Adds ```max_token_length``` option to the ```CharGroupTokenizer```.
Updates documentation as well to reflect the changes.

Solves #56676

### CODE REVIEW

File: modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CharGroupTokenizerFactory.java [hunk 1]
```diff
@@ -41,6 +46,8 @@
     public CharGroupTokenizerFactory(IndexSettings indexSettings, Environment environment, String name, Settings settings) {
         super(indexSettings, settings, name);
 
+        maxTokenLength = settings.getAsInt(MAX_TOKEN_LENGTH, StandardAnalyzer.DEFAULT_MAX_TOKEN_LENGTH);
```
Comment (cbuescher) [inline modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CharGroupTokenizerFactory.java:49]: Super small nitpick: can you change to use the default from CharTokenizer.DEFAULT_MAX_WORD_LEN here? Its the same value (255) as `StandardAnalyzer.DEFAULT_MAX_TOKEN_LENGTH` currently, just so we tie the default to the one thats used in the class we actually use.

**Review by cbuescher — APPROVED** [review]
Hi @ADBalici, thanks for opening this PR, looks great. I left one super small comment but I'd also merge this if you don't get to changing it. Let me know, I will need to run out CI tests on this before merging in any case.

**Comment by cbuescher** [comment]
@elasticmachine test this please

File: modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CharGroupTokenizerFactoryTests.java [hunk 2]
```diff
@@ -61,6 +64,52 @@ public void testParseTokenChars() {
...
+        final Settings indexSettings = newAnalysisSettingsBuilder().build();
+        IndexSettings indexProperties = IndexSettingsModule.newIndexSettings(index, indexSettings);
+        final String name = "cg";
+
+        String[] conf = new String[] {"-"};
+
+        final Settings defaultLengthSettings = newAnalysisSettingsBuilder()
+            .putList("tokenize_on_chars", conf)
+            .build();
+        CharTokenizer tokenizer = (CharTokenizer) new CharGroupTokenizerFactory(indexProperties, null, name, defaultLengthSettings).create();
```
Comment (cbuescher) [inline modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CharGroupTokenizerFactoryTests.java:78]: nit: we have a 140 chars line length limit that gets checked via "checkstyle" in CI and it complains about this line being too long. Simply splitting this is fine.

**Comment by ADBalici** [comment]
Thanks for reviewing this @cbuescher 
All points that you raised were addressed. :smile:

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@ADBalici I took the liberty to change another checkstyle issue that appeared when running the test, hope you don't mind
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by ADBalici** [comment]
it seems that some of the checks are failing, althgough it's not obvious to my why

**Comment by cbuescher** [comment]
@ADBalici test are all good now, I merged to master and the upcoming 7.9 branch. Thanks for your contribution!

**Pull request merged**